### DISCUSSION TITLE
Forking Renovate, why re-open old config error issue, and why no config warning on Dependency Dashboard?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#29132] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Forking Renovate, GitHub, do not know what version

### Please tell us more about your question or problem

I commented on a PR on another repository to answer some questions from a user about their Renovate config:

- https://github.com/slsa-framework/slsa-github-generator/pull/3635#issuecomment-2115086864

The member of the repository asked me some more questions based on my comment:

- https://github.com/slsa-framework/slsa-github-generator/pull/3635#issuecomment-2116420633

I don't know the answer to two of their questions. So I'm asking them here:

- [x] Why does Forking Renovate re-open a old config error issue: https://github.com/slsa-framework/slsa-github-generator/issues/404, instead of creating a _new_ config error issue? This effectively "hides" the old issue behind newer issues, because GitHub defaults to show the highest (newest) issue numbers first.
- [x] Why does the Forking Renovate Dependency Dashboard not warn about their config error? https://github.com/slsa-framework/slsa-github-generator/issues/408 Check the edit history, got to `forking-renovate bot edited on Dec 6, 2023` to a `WARN: Package lookup failures` message, but _no config error message_.

Luckily I managed to answer at least one of their questions. :upside_down_face: 

### Logs (if relevant)

<details><summary>No logs, as I don't have access to their logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
In the old days we had a few incidents where Renovate kept making issues over and over, so I defaulted behavior to reopen/reuse the old issue to keep the noise lower. It's maybe time to instead stop the reopening

**Reply by HonkingGoose** [reply]
We have a [`configWarningReuseIssue` config option](https://docs.renovatebot.com/configuration-options/#configwarningreuseissue):

> ## `configWarningReuseIssue`
> 
> Set this to `false` to make Renovate create a new issue for each config warning, instead of reopening or reusing an existing issue.
> 
> | Name | Value |
> |--------|--------|
> | type | boolean |
> | default | `true` |
> | cli | `--config-warning-reuse-issue` |
> | env | `RENOVATE_CONFIG_WARNING_REUSE_ISSUE` | 
> 
> Renovate's default behavior is to reuse/reopen a single Config Warning issue in each repository so as to keep the "noise" down. However for some people this has the downside that the config warning won't be sorted near the top if you view issues by creation date. Configure this option to `false` if you prefer Renovate to open a new issue whenever there is a config warning.

I'm opening a PR on their repository to set `configWarningReuseIssue` to `false`, so Renovate matches their expectations:

- https://github.com/slsa-framework/slsa-github-generator/pull/3662

## Issue to change default behavior in the `v38` major release?

Changing the default behavior could go in the major `v38` release. Do you want me to create an issue then, for us to track the work?

The issue would be labeled as a breaking change and for `v38`, and have these todos:

- default `configWarningReuseIssue` to `false`, so Renovate opens a new issue by default
- update tests
- update the docs
- write release note entry in the v38 changelog, that explain why we're making the change, and that users can opt-out by setting `configWarningReuseIssue` to `true` so Renovate re-opens the old issue

**Reply by HonkingGoose** [reply]
@rarkins can you read the comment above this one, and tell me if you want an issue to change the default behavior, so that Renovate creates a new config warning issue? That change could go in the new `v38` release maybe?

**Reply by rarkins** [reply]
Yes, let's change the default in v38

**Reply by HonkingGoose** [reply]
See issue:

- #29504

**Comment by HonkingGoose** [comment, marked as answer]
I'm closing this Discussion as resolved, because:

- the `slsa-framework/slsa-github-generator` repository now has a working Renovate config with a bit of help from `@rarkins` and me
- we have an issue to change the default behavior of `configWarningReissue` to `false` in a upcoming major release: #29504
